src/Octoshift/GithubClient.cs

```diff
@@ -82,7 +88,14 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url)
             var content = await response.Content.ReadAsStringAsync();
             _log.LogVerbose($"RESPONSE ({response.StatusCode}): {content}");
 
-            response.EnsureSuccessStatusCode();
+            if (shouldEnsureSuccessStatusCode)
```

Comment:
@dylan-smith I added a comment about refactoring the `GithubApi`, but even in that case we are going to need this flag b/c success is a range rather than just OK, i.e. 2XX

Reply:
either way suggested I think works fine, in my suggestion I think it does still work, because by comparing it against OK (which is the default assigned to that arg) you are basically checking if the caller did NOT pass in an explicit status value, and if not calling EnsureSuccess().

Reply:
We want `shouldEnsureSuccessStatusCode` because the API returns a `302` found instead of a 200. Which is the correct response for what we're trying to do

Reply:
right, but what I'm saying is if you just pass the status arg (and eliminate the shouldEnsureSuccessStatusCode arg) you can just check if status is set, and if so skip the EnsureSuccess() call and check if it equals 302 (or whatever status is set to)

Reply:
I see, thanks for explaining. It looks like there's no null type for HttpStatusCodes, so it would be checking against the default value set, probably `OK`

Reply:
I still don't think comparing against 200 would be a good idea since for example `201` also indicates success!.

Reply:
As written (this line is out of date), the if the default status hasn't been changed (`OK` 200) then it uses `ensuresuccessstatuscode` which checks 200-299 (it does not check value content). If the default status has changed, it validates against that :+1: